src/plugins/discover/public/application/main/hooks/use_discover_state.ts

```diff
@@ -194,10 +194,10 @@ export function useDiscoverState({
    */
   useEffect(() => {
     const unsubscribe = appStateContainer.subscribe(async (nextState) => {
-      const { hideChart, interval, sort, index } = state;
-      // chart was hidden, now it should be displayed, so data is needed
-      const chartDisplayChanged = nextState.hideChart !== hideChart && hideChart;
+      const { hideChart, interval, breakdownField, sort, index } = state;
+      const chartDisplayChanged = nextState.hideChart !== hideChart;
       const chartIntervalChanged = nextState.interval !== interval;
+      const breakdownFieldChanged = nextState.breakdownField !== breakdownField;
```

Comment:
Breakdown field changes trigger a refetch in unified histogram, and need to in Discover as well to keep in sync.